src/client/kube_store/index.ts

```diff
@@ -233,6 +48,47 @@ export default class KubeStore {
     return this.instance || (this.instance = new this(''));
   }
 
+  public patchResource(resource: KubeResource, name: string, patch: object): object {
+    let source: object = {};
+    let key: string = '';
+    const myPatch: object = {'foo':'bar'};
+    if (resource instanceof NamespacedResource) {
+      const namespacedResource = resource as NamespacedResource;
+      if (resource instanceof CoreNamespacedResource) {
+        // Core Namespaced
+        key = `api/${resource.gvk().version}/namespaces/${namespacedResource.namespace}/${resource.gvk().kindPlural}`;
+        if (this.data()[key]) {
+          source = this.data()[key][name];
+        }
+      } else {
+        // Extension Namespaced
+        key = `apis/${resource.gvk().group}/${resource.gvk().version}` +
+          `/namespaces/${namespacedResource.namespace}/${resource.gvk().kindPlural}`;
+        if (this.data()[key]) {
+          source = this.data()[key][name];
+        }
+      }
+    } else {
+      if (resource instanceof CoreClusterResource) {
+        // Core ClusterResource
+        key = `api/${resource.gvk().version}/namespaces/${resource.gvk().kindPlural}`;
+        if (this.data()[key]) {
+          source = this.data()[key][name];
+        }
+      } else {
+        // Extension ClusterResource
+        key = `apis/${resource.gvk().group}/${resource.gvk().version}/${resource.gvk().kindPlural}`;
+        if (this.data()[key]) {
+          source = this.data()[key][name];
+        }
+      }
+    }
+    const patchedObject = JsonMergePatch.apply(source, myPatch);
```

Comment:
assume this should be 

const patchedObject = JsonMergePatch.apply(source, patch);